plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorSmokeTest.java

```diff
@@ -107,4 +112,63 @@ public void testS3LocationWithTrailingSlash()
 
         assertUpdate("DROP TABLE " + tableName);
     }
+
+    @Test
+    public void testExpireSnapshots()
+    {
+        String tableName = "test_expiring_snapshots_" + randomTableSuffix();
+        Session sessionWithShortRetentionUnlocked = prepareCleanUpSession();
+        String location = "s3://%s/%s/%s/".formatted(bucketName, schemaName, tableName);
+        List<Event> events = new ArrayList<>();
+        hiveMinioDataLake.getMinioClient().captureBucketNotifications(bucketName, event -> {
+            if (event.eventType().toString().toLowerCase(ENGLISH).contains("remove")) {
+                events.add(event);
+            }
+        });
+
+        assertUpdate("CREATE TABLE " + tableName + " (key varchar, value integer) WITH (location='" + location + "')");
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('one', 1)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('two', 2)", 1);
+        assertThat(query("SELECT sum(value), listagg(key, ' ') WITHIN GROUP (ORDER BY key) FROM " + tableName))
+                .matches("VALUES (BIGINT '3', VARCHAR 'one two')");
+
+        List<String> initialMetadataFiles = hiveMinioDataLake.getMinioClient().listObjects(bucketName, "/%s/%s/metadata".formatted(schemaName, tableName));
+        assertThat(initialMetadataFiles).isNotEmpty();
+
+        List<Long> initialSnapshots = getSnapshotIds(tableName);
+        assertThat(initialSnapshots).isNotEmpty();
+
+        assertQuerySucceeds(sessionWithShortRetentionUnlocked, "ALTER TABLE " + tableName + " EXECUTE EXPIRE_SNAPSHOTS (retention_threshold => '0s')");
+
+        List<String> updatedMetadataFiles = hiveMinioDataLake.getMinioClient().listObjects(bucketName, "/%s/%s/metadata".formatted(schemaName, tableName));
+        assertThat(updatedMetadataFiles).isNotEmpty().hasSizeLessThan(initialMetadataFiles.size());
+
+        List<Long> updatedSnapshots = getSnapshotIds(tableName);
+        assertThat(updatedSnapshots.size()).isLessThan(initialSnapshots.size());
+        assertThat(updatedSnapshots.size()).isEqualTo(1);
+
+        assertThat(query("SELECT sum(value), listagg(key, ' ') WITHIN GROUP (ORDER BY key) FROM " + tableName))
+                .matches("VALUES (BIGINT '3', VARCHAR 'one two')");
+        assertThat(events.stream()
+                .map(event -> Integer.parseInt(event.responseElements().get("content-length"))).filter(size -> size > 0).collect(toImmutableList()))
+                .hasSize(2);
```

Comment:
why 2? why `size > 0`?

document.
also recognizing batch by content-length looks very indirect. is there anything that looks more direct? eg what's the content?

Reply:
maybe instead of this -- have the table with 10 snapshots and verify there is  only deletion request to MinIO, not 10.

Reply:
ehh I thought I explained that during offline discussion. I can create 100 snapshots and there will be 100 events registered by Minio. I didn't manage to see incoming requests I can only see events that are registered and based on that I was not able to find a better solution than this one based on content-length. I was not able to access content itself.

BUT i found a little better way now :D